### PR TITLE
Enhance KDUMP testcase to check VM readiness by probing instead of using fixed time

### PR DESCRIPTION
**Pull request by johnsongeorge-w**

### CODE REVIEW

File: Testscripts/Windows/KDUMP-TestScript.ps1 [hunk 1]
```diff
@@ -115,14 +151,16 @@ function Main {
...
+
     # Rebooting the VM in order to apply the kdump settings
     Run-LinuxCmd -username $VMUserName -password $VMPassword -ip $Ipv4 -port $VMPort `
         -command "sync; reboot" -runAsSudo -RunInBackGround | Out-Null
     Write-LogInfo "Rebooting VM $VMName after kdump configuration..."
-    Start-Sleep -Seconds 10 # Wait for kvp & ssh services stop
 
     # Wait for VM boot up and update ip address
-    Wait-ForVMToStartSSH -Ipv4addr $Ipv4 -StepTimeout 360 | Out-Null
+    Wait-ForVMToStartSSH -Ipv4addr $Ipv4 -StepTimeout $waitToStartTime | Out-Null
```
Comment (LiliDeng) [inline Testscripts/Windows/KDUMP-TestScript.ps1:164]: after extending the time, not enough?

Comment (johnsongeorge-w) [reply Testscripts/Windows/KDUMP-TestScript.ps1:164]: No. Again I added a logic to make sure VM booted up without waiting for fixed time.

Comment (LiliDeng) [reply Testscripts/Windows/KDUMP-TestScript.ps1:164]: does the 22 open but VM still can't be login?

Comment (johnsongeorge-w) [reply Testscripts/Windows/KDUMP-TestScript.ps1:164]: Yes

Comment (LiliDeng) [reply Testscripts/Windows/KDUMP-TestScript.ps1:164]: I have a test, we have known bug when trigger crash on some specified CPU range, in this case, we have to wait till timeout.

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**